server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -0,0 +1,260 @@
+        final Integer start = request.getStartPosition();
+        final Integer end = request.getEndPosition();
+        final MotifSearchType searchType = request.getSearchType();
+        if (searchType.equals(MotifSearchType.REGION)) {
+            Assert.notNull(request.getChromosomeId(), getMessage("Chromosome not provided!"));
+            Assert.notNull(start, getMessage("Start position is empty!"));
+            Assert.notNull(end, getMessage("End position is empty!"));
+            Assert.isTrue(end - start > 0,
+                    getMessage("Provided end and start are not valid: " + end + " < " + start));
+        } else if (searchType.equals(MotifSearchType.CHROMOSOME)) {
+            Assert.notNull(request.getChromosomeId(), getMessage("Chromosome not provided!"));
+            if (end != null && start != null) {
+                Assert.isTrue(end - start > 0,
+                        getMessage("Provided end and start are not valid: " + end + " < " + start));
+            }
+        } else if (searchType.equals(MotifSearchType.WHOLE_GENOME)) {
+            Assert.notNull(request.getReferenceId(), getMessage("Genome id is empty!"));
+        }
+    }
+
+    private MotifSearchResult searchRegionMotifs(final MotifSearchRequest request) {
+        final Chromosome chromosome = loadChrById(request.getReferenceId(), request.getChromosomeId());
+        return MotifSearchResult.builder()
+                .result(
+                        fillMotifList(
+                                chromosome, request.getStartPosition(),
+                                request.getEndPosition(), 0,
+                                request.getStrand()))
+                .chromosomeId(request.getChromosomeId())
+                .pageSize(request.getPageSize())
+                .position(request.getEndPosition())
+                .build();
+    }
+
+    private MotifSearchResult searchChromosomeMotifs(final MotifSearchRequest request) {
+        final Chromosome chromosome = loadChrById(request.getReferenceId(), request.getChromosomeId());
+        int start = request.getStartPosition() == null
+                ? 0
+                : request.getStartPosition();
+        int end = request.getEndPosition() == null
+                ? chromosome.getSize()
+                : request.getEndPosition();
+        return MotifSearchResult.builder()
+                .result(fillMotifList(
+                        chromosome, start, end,
+                        request.getPageSize(),
+                        request.getStrand()))
+                .chromosomeId(request.getChromosomeId())
+                .pageSize(request.getPageSize())
+                .position(end)
+                .build();
+    }
+
+    private MotifSearchResult searchWholeGenomeMotifs(final MotifSearchRequest request) {
+        MotifSearchRequest currentRequest;
+        int pageSize = request.getPageSize() == null ? 0 : request.getPageSize();
```

Comment:
lets introduce a constant which we will use of page size is not present, it is strange to use 0